nodebuilder/store.go

```diff
@@ -150,6 +156,73 @@ type fsStore struct {
 	dirLock *flock.Flock // protects directory
 }
 
+// DiscoverOpened finds a path of an opened Node Store and returns its path.
+// If multiple nodes are running, it only returns the path of the first found node.
+// Network is favored over node type.
+//
+// Network preference order: Mainnet, Mocha, Arabica, Private
+// Type preference order: Bridge, Full, Light
+func DiscoverOpened() (string, error) {
+	defaultNetwork := []p2p.Network{p2p.Mainnet, p2p.Mocha, p2p.Arabica, p2p.Private}
+	nodeTypes := []nodemod.Type{nodemod.Bridge, nodemod.Full, nodemod.Light}
```

Comment:
I'd recommend modifying `listProvidedNetworks` from `github.com/celestiaorg/celestia-node/nodebuilder/p2p`, making it public, and getting the list of networks altogether(the same with network types). Otherwise, we can forget to include them here, once we'll add new networks/types.

Reply:
I added in a slice to handle the ordering since apparently, maps randomize order when iterating. Also, added support for custom networks now too.